BusinessLayerTests/InventoryCsvManagerTests.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Extensions.Configuration;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Models;
+using Moq;
+using Serilog;
+
+namespace BusinessLayer.Tests
+{
+    [TestClass]
+    public class InventoryCsvManagerTests
+    {
+        private InventoryCsvManager _inventoryCsvManager;
+        private IConfigurationRoot _mockedConfig;
+
+        private List<Contact> contacts;
+
+        [TestInitialize]
+        public void Initialize()
+        {
+            var configMock = new Mock<IConfigurationRoot>();
+            configMock.SetupGet(x => x["INVENTORY_PATH"])
+                .Returns(Path.Combine(Directory.GetCurrentDirectory(), "inventory.csv"));
+            _mockedConfig = configMock.Object;
+            var mockLog = new Mock<ILogger>().Object;
+            _inventoryCsvManager = new InventoryCsvManager(_mockedConfig, mockLog);
+        }
+
+        [TestMethod]
+        public void GetLastSerialNumber_ShouldReturnInt()
+        {
+            Assert.IsInstanceOfType(_inventoryCsvManager.GetLastSerialNumber(), typeof(long));
+        }
+
+        [TestMethod]
+        public void AddSerialNumber_ShouldAddTheCorrectSerialNumber()
+        {
+            var contact = new Contact {Name = "Jean", Mail = "marcelle@jean.be"};
+
+            _inventoryCsvManager.AddSerialNumber(contact);
+            var _reverseStream = new ReverseLineReader(_mockedConfig["INVENTORY_PATH"]).Take(2);
```

Comment:
this is not a unit test

Reply:
Integration Test ?
This "test" add a serial number to a file, and then read the last entry to check if it's the correct one.
Need to be refactored or deleted ?